jtravis/src/main/java/fr/inria/spirals/jtravis/helpers/BuildHelper.java

```diff
@@ -102,112 +102,171 @@ private static boolean isAcceptedBuild(Build build, int prNumber, BuildStatus st
         return true;
     }
 
-    /**
-     * This is a recursive method allowing to get build from a slug.
-     *
-     * @param slug The slug where to get the builds
-     * @param result The list result to aggregate
-     * @param limitDate If given, the date limit to get builds: all builds *before* this date are considered
-     * @param after_number Used for pagination: multiple requests may have to be made to reach the final date
-     * @param eventTypes Travis support multiple event types for builds like Push, PR or CRON. Those are retrieved individually
-     * @param limitNumber Allow to finish early based on the number of builds selected. if 0 passed use only the date to stop searching
-     * @param status Allow to only select builds of that status, if null it takes all status
-     * @param prNumber Allow to only consider builds of that PR, if -1 given it takes all builds
-     * @param onlyAfterNumber Consider only builds after the specified after_number: should be use in conjunction with a specified after_number.
-     */
-    private static void getBuildsFromSlugRecursively(String slug, List<Build> result, Date limitDate, int after_number, List<String> eventTypes, int limitNumber, BuildStatus status, int prNumber, boolean onlyAfterNumber, String previousBranch) {
-        Map<Integer, Commit> commits = new HashMap<Integer,Commit>();
-
+    private static String getResourceUrl(String slug, String eventType, int after_number) {
         String resourceUrl = getInstance().getEndpoint()+RepositoryHelper.REPO_ENDPOINT+slug+"/"+BUILD_NAME;
 
-        if (after_number <= 0 && onlyAfterNumber) {
-            return;
-        }
-
-        if (eventTypes.isEmpty()) {
-            return;
+        if (eventType != null) {
+            resourceUrl += "?event_type=" + eventType;
         }
-        String evenType = eventTypes.get(0);
-        resourceUrl += "?event_type="+evenType;
-
         if (after_number > 0) {
-            resourceUrl += "&after_number="+after_number;
+            if (eventType != null) {
+                resourceUrl += "&after_number=" + after_number;
+            } else {
+                resourceUrl += "?after_number="+after_number;
+            }
         }
 
-        boolean dateReached = false;
+        return resourceUrl;
+    }
 
+    private static void getBuildsAndCommits(String resourceUrl, JsonArray buildArray, Map<Integer, Commit> commits) {
```

Comment:
I really don't like the buildArray parameter which is used to return a value... Why not returning it directly?